cmd/snap-update-ns/mount-entry.c

```diff
@@ -114,11 +114,15 @@ int sc_compare_mount_entry(const struct sc_mount_entry *a,
 	if (a == NULL || b == NULL) {
 		die("cannot compare NULL mount entry");
 	}
-	result = strcmp(a->entry.mnt_fsname, b->entry.mnt_fsname);
+	// NOTE: sort reorder field so that mnt_dir is before
+	// mnt_fsname. This ordering is a little bit more interesting
+	// as the directory matters more and allows us to do useful
+	// things later.
+	result = strcmp(a->entry.mnt_dir, b->entry.mnt_dir);
```

Comment:
I have the hunch that you need to split on / and sort the paths as tuples to get correct behavior in sc_compute_required_mount_changes or you need to keep track of multiple prefixes

consider this directories list for example (just sorting the strings):

['/a/b', 'a/b-1', 'a/b-1/foo', 'a/b/c']